system-addon/lib/TelemetryFeed.jsm

```diff
@@ -119,7 +118,10 @@ this.TelemetryFeed = class TelemetryFeed {
       return;
     }
 
-    session.session_duration = Math.round(Components.utils.now() - session.start_time);
+    if (session.perf.visibility_event_rcvd_ts) {
+      session.session_duration = perfService.absNow() - session.perf.visibility_event_rcvd_ts;
```

Comment:
Is it possible to stick with the millisecond precision for `session_duration`? Unfortunately, our data warehouse doesn't support altering the column type (int -> double) on the fly. To do that regardless would need a series of table migrations.

@emtwo Could you also confirm this?

Making `session_duration` as an optional field is fine, we can handle it in the ETL job.